aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -1299,16 +1277,41 @@ TORCH_IMPL_FUNC(scatter_add)
   }
 }
 
-TORCH_IMPL_FUNC(_scatter_reduce_structured_cpu)
-(const Tensor& self,
- int64_t dim,
- const Tensor& index,
- const c10::string_view reduce,
- const c10::optional<int64_t> output_size,
- const Tensor& out) {
+Tensor scatter_reduce_two_cpu(const Tensor& self,
+                              int64_t dim,
+                              const Tensor& index,
+                              const c10::string_view reduce,
+                              const c10::optional<int64_t> output_size,
+                              const c10::optional<Tensor>& optional_out) {
```

Comment:
just want to double check that I didn't misunderstand - by making `scatter_reduce` unstructured and removing the out variant that means we still want the optional_out argument from `torch_scatter` so gradients can be calculated, is that correct? If not I'll remove this additional argument and undo the changes!

Reply:
Let's remove this argument for now and introduce it later on. Our common definition of out variants doesn't support gradients and we shouldn't let this one-off operator intersect with that.